### ISSUE TITLE
I was trying to migrate to SDL2 this game too.

### ISSUE THREAD

**Issue by mdtrooper** [petitlapin/Li-Ri#25]
Hi.

You are "fast cowboy of the western film" and I glad for it.

For to give some context:

https://gitlab.com/LibreGames/libregames.gitlab.io/-/issues/4

https://gitlab.com/tres-14159/ri-li

Could I help in this project?

Regards.

**Comment by petitlapin** [comment]
Hi!
I mostly had the same approach (in French but it explains why and what I did: https://linuxfr.org/users/johnny_jazeix/journaux/li-ri-fork-et-portage-de-ri-li-sous-android).

I tried to contact the original author but didn't managed, so I forked and renamed it. As said in your thread, I should have kept the original history (at least tried to have the versions) but I didn't think of it at the time and started from the zip files in https://ri-li.sourceforge.net/download.html and https://ri-li.sourceforge.net/devel.html.

I first ported it to SDL2, then Android (which is mostly my main goal) and recently it's mostly code cleanup (translation in English, tidying, formatting), helping in case of packaging and setting up as much automation as I can: generation of the changelog (in the different places to not duplicate them)/change of version via scripts...

I don't really plan to add too much features myself, just fixing the issues that are raised and continue cleaning the code so it can be more easily maintainable.
One point I didn't have to check is the internationalization. For now, the translated texts are handled via images, which is clearly not good (I don't even know how to generate the images and add a new translation...). I would like at some point to take a look at adding po files, but I fear we might need to add SDL_ttf dependency and find fonts (I don't know which fonts were actually used to generate the current texts).

It may make sense to fork this project in LibreGames and continue the development there (I can create an account there and develop there) but there are a few questions we have to think about:
* What about the name? I changed to Li-ri because I didn't want to keep Ri-Li without the original author approval. So I would prefer to keep it that way if possible.
* I used github actions to generate the different packages for Linux, Windows and MacOS (not tested) and other various job. We can obviously do the same within Gitlab (we do it in KDE) but it's maybe something to set up and can take some time we want to have the same checks.
* I have to check with f-droid folks if we can rename it (if we move it, it would make more sense to rename the package to org.libregames.liri and change the upstream repository but I don't know if it is doable).
* Would it be possible to then restart from a clean git fork from the original repository (with the history) and reapply all the commits from here so we have a really clean history with all the versions?

So to sum up, I'm glad we have the same idea and if we can move it and continue the development in collaboration in LibreGames, that's good for me :).

Cheers!

**Comment by petitlapin** [comment]
I've discussed with f-droid people, we can rename the apps, but that would mean all users need to reinstall the app (and we don't really have a way to tell them).

So we can do it, but if we change it, we have to do it the sooner we can :). Can we involve LibreGames people here so we can quickly decide if we plan to migrate the code there, and we can first rename, then discuss all the other steps?

**Comment by mdtrooper** [comment]
I think that migrate this project is not necessary. It is correct that Li-Ri is on gihtub. Don't move it.

I am in "debian-devel-games" mailing list (it is not especial, everybody can subscribe it). But I am going to write about your repo to get this code for the Ri-Li package.

Well, do we (me and you) close this issue?

**Comment by petitlapin** [comment]
Great, if you need to upstream some info (patch, debian package files...), don't hesitate to send a PR or just the links, I'll add them to this repository!

**Comment by akien-mga** [comment]
From my side, if you ever need a "collective" home for the project I'm happy to add more people to LibreGames. But we're a small group and mostly dormant right now, as life and other projects got in the way of the ideal of keeping open source games up to date :)

So I think it makes sense to keep maintaining this project here, especially if it has an extensive GitHub Actions setup.

I'm also the `ri-li` packager for Mageia, I'll have a look at packaging `li-ri` as a replacement.

**Comment by petitlapin** [comment]
ok, then let's keep it here :).
When I'll be less active, maybe it will be a good time to migrate it LibreGames. I'm also quite involved in KDE so I totally understand how it's difficult to juggle between life and multiple projects :).

I'll need to create a new version before you do a package as the last official version crashes when exiting Li-Ri. I have fixed it but haven't yet made a tag.
